pkg/infrastructure/powervs/clusterapi/powervs.go

```diff
@@ -438,5 +452,38 @@ func (p Provider) PostProvision(ctx context.Context, in clusterapi.PostProvision
 		return fmt.Errorf("failed to add SSH key for the workers(%s): %w", fieldType, err)
 	}
 
+	// Step 3.
+	// Add the bootstrap's IP address to the load balancer pool
+	// Get the cluster from the provider so we can have what load balancers are attached
+	powerVSCluster := &capibm.IBMPowerVSCluster{}
+	key = crclient.ObjectKey{
+		Name:      in.InfraID,
+		Namespace: capiutils.Namespace,
+	}
+	logrus.Debugf("PostProvision: cluster key = %+v", key)
+	if err = in.Client.Get(ctx, key, powerVSCluster); err != nil {
+		return fmt.Errorf("failed to get PowerVS cluster in PostProvision: %w", err)
+	}
+
+	lbIntExp := regexp.MustCompile(`\b-loadbalancer-int\b$`)
+
+	// Find the internal load balancer
+	for lbKey, loadBalancerStatus := range powerVSCluster.Status.LoadBalancers {
+		if lbIntExp.MatchString(lbKey) {
+			logrus.Debugf("PostProvision: Found internal load balancer ID = %s, State = %s, Hostname = %s",
+				*loadBalancerStatus.ID,
+				loadBalancerStatus.State,
+				*loadBalancerStatus.Hostname)
+
+			if err = client.AddIPToLoadBalancerPool(ctx,
+				*loadBalancerStatus.ID,
+				"additional-pool-22623",
+				22623,
+				ipAddr); err != nil {
+				return fmt.Errorf("failed to add the bootstrap IP to the load balancer pool: %w", err)
+			}
+		}
```

Comment:
You can decrese the indent level here if you `continue` when there is no match.
```suggestion
		if !lbIntExp.MatchString(lbKey) {
		    continue
		}
		logrus.Debugf("PostProvision: Found internal load balancer ID = %s, State = %s, Hostname = %s",
			*loadBalancerStatus.ID, loadBalancerStatus.State, *loadBalancerStatus.Hostname)

		if err = client.AddIPToLoadBalancerPool(ctx, *loadBalancerStatus.ID, "additional-pool-22623", 22623, ipAddr); err != nil {
			return fmt.Errorf("failed to add the bootstrap IP to the load balancer pool: %w", err)
		}
```

Reply:
Done